strawberry/schema/schema.py

```diff
@@ -36,27 +36,34 @@ def __init__(
         execution_context_class: Optional[Type[GraphQLExecutionContext]] = None,
     ):
         self.extensions = extensions
-        self.type_map: Dict[str, ConcreteType] = {}
+
         self.execution_context_class = execution_context_class
 
-        query_type = get_object_type(query, self.type_map)
-        mutation_type = get_object_type(mutation, self.type_map) if mutation else None
+        self.schema_converter = GraphQLCoreConverter()
+
+        query_type = self.schema_converter.from_object_type(query)
+        mutation_type = (
+            self.schema_converter.from_object_type(mutation) if mutation else None
+        )
         subscription_type = (
-            get_object_type(subscription, self.type_map) if subscription else None
+            self.schema_converter.from_object_type(subscription)
+            if subscription
+            else None
         )
```

Comment:
Hmm it might make the code here visually cleaner, but I'm not sure it makes sense from the design of the schema converter. What would it really mean if we asked it to convert `None` to an GraphQL object? I'm not sure short circuiting a None-in-None-out situation will play nicely into the future continued abstraction of the converter.

Reply:
got it, I can live with that :D